chain/client/src/chunk_validation.rs

```diff
@@ -486,24 +486,107 @@ impl Client {
         Ok((main_transition, implicit_transitions, receipts_hash))
     }
 
+    /// State witness proves the execution of receipts proposed by `prev_chunk`.
+    /// This function collects all incoming receipts for `prev_chunk`, along with the proofs
+    /// that those receipts really originate from the right chunks.
+    /// TODO(resharding): `get_incoming_receipts_for_shard` generates invalid proofs on resharding
+    /// boundaries, because it removes the receipts that target the other half of a split shard,
+    /// which makes the proof invalid. We need to collect the original proof and later, after verifcation,
+    /// filter it to remove the receipts that were meant for the other half of the split shard.
+    fn collect_source_receipt_proofs(
+        &self,
+        prev_block: &Block,
```

Comment:
`prev_block_hash` should be enough here, so you don't have to require the whole block in `send_chunk_state_witness_to_chunk_validators`